Fakebook.Profile/Fakebook.Profile.RestApi/Controllers/ProfileController.cs

```diff
@@ -0,0 +1,167 @@
+        /// <returns>A collection of profiles converted to API Models</returns>
+        [HttpGet("selection/{emails}")]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public async Task<ActionResult<IEnumerable<ProfileApiModel>>> SelectProfilesAsync([FromBody] IEnumerable<string> emails)
+        {
+            var results = await _repository.GetProfilesByEmailAsync(emails);
+
+            // convert them to the ApiModel
+            return Ok(results
+                .Select(p => new ProfileApiModel(p))
+                .ToList());
+        }
+
+        /// <summary>
+        /// Action method that handles getting a single user by their email;
+        /// GET: /api/profiles/{profileEmail}
+        /// </summary>
+        /// <param name="profileEmail">The email of the user being retrieved</param>
+        /// <returns>If found, a profile API model version of the profile; if not, it returns a NotFound()</returns>
+        [HttpGet("")]
+        [HttpGet("{profileEmail}")]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public async Task<ActionResult<ProfileApiModel>> GetAsync(string profileEmail = null)
+        {
+            // redundant or incorrect??
+            string email = profileEmail is not null ? profileEmail : GetUserEmail();
+
+            if (email is null)
+            {
+                _logger.LogError($"Could not find current user's email. {profileEmail}");
+                throw new ArgumentException("Could not find current user's email");
+            }
+
+            var result = await _repository.GetProfileAsync(email);
+            return Ok(new ProfileApiModel(result));
+        }
+
+        /// <summary>
+        /// Action method that is used for creating a new profile;
+        /// POST: /api/profiles/
+        /// </summary>
+        /// <param name="apiModel">The data of the profile to be created</param>
+        /// <returns>Created if the model was created successfully, otherwise a 400-based status code</returns>
+        [Authorize]
+        [HttpPost]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status201Created)]
+        public async Task<ActionResult> CreateAsync([FromBody] ProfileApiModel apiModel)
+        {
+            try
+            {
+                var domainProfile = apiModel.ToDomainProfile();
+                await _repository.CreateProfileAsync(domainProfile);
+                return CreatedAtAction(nameof(GetAsync), new { email = apiModel.Email });
+            }
+            catch (Exception e)
```

Comment:
not all exceptions indicate a problem with the request.
for logging exceptions, just use the overloads that log the entire exception object.